lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -608,20 +667,51 @@
         return toReturn;
     }());
 
+    // mappedDefaultLocale will get the default locale and update any deprecated
+    // collation/sort order values it may use
+    let __mappedDefaultLocale = undefined;
     var mappedDefaultLocale = function () {
-        var parts = platform.builtInRegexMatch(GetDefaultLocale(), /([^_]*)_?(.+)?/);
-        var locale = parts[1];
-        var collation = parts[2];
-        var availableBcpTags = reverseLocaleAcceptingCollationValues[locale];
+        if (__mappedDefaultLocale) {
+            return __mappedDefaultLocale;
+        }
 
-        if (collation === undefined || availableBcpTags === undefined) { return locale; }
+        let locale = undefined;
+        let collation = undefined;
+        if (isPlatformUsingICU) {
+            // ICU's getDefaultLocale() will return a valid BCP-47/RFC 5646 langtag
+            locale = GetDefaultLocale();
+            const match = platform.builtInRegexMatch(locale, /-u(-[^\-][^\-]?-[^\-]+)*-co-([^\-]+).*/);
+            if (match) {
+                collation = match[2];
+                locale = callInstanceFunc(StringInstanceReplace, locale, `-co-${match[2]}`, "");
+            }
+        } else {
+            // Windows' getDefaultLocale() will return a weird RFC4646 langtag
+            const parts = platform.builtInRegexMatch(GetDefaultLocale(), /([^_]*)_?(.+)?/);
+            locale = parts[1];
+            collation = parts[2];
+        }
+
+        if (collation === undefined) {
+            __mappedDefaultLocale = locale;
+            return __mappedDefaultLocale;
+        }
 
-        var bcpTag = availableBcpTags[collation];
-        if (bcpTag !== undefined) {
-            return locale + "-u-co-" + bcpTag;
+        const collationMapForLocale = reverseLocaleAcceptingCollationValues[locale];
+        if (collationMapForLocale === undefined) {
+            // Assume the system wouldn't give us back a bad collation value
+            __mappedDefaultLocale = `${locale}-u-co-${collation}`;
+            return __mappedDefaultLocale;
         }
 
-        return locale;
+        const mappedCollation = collationMapForLocale[collation];
+        if (mappedCollation !== undefined) {
+            __mappedDefaultLocale = `${locale}-u-co-${mappedCollation}`;
+        } else {
+            __mappedDefaultLocale = `${locale}-u-co-${collation}`;
```

Comment:
Same -- have you exercised these paths? The Intl harness should be helpful here.